spec/controllers/students_controller_spec.rb

```diff
@@ -43,11 +43,15 @@
         let(:params) { { course_id: course.id } }
       end
 
-      it 'calls perform_later on a background job' do
-        expect(UploadRolesJob).to receive(:perform_later).and_return OpenStruct.new(job_id: 1)
-        post_as instructor,
-                :upload,
-                params: { course_id: course.id, upload_file: fixture_file_upload('students/form_good.csv', 'text/csv') }
+      ['.csv', '', '.pdf'].each do |extension|
+        ext_string = extension.empty? ? 'none' : extension
+        it "calls perform_later on a background job on a valid file with extension #{ext_string}" do
```

Comment:
oh I realized here (and in all other descriptions) that you should use `"#{ext_string}"` rather than just `#{ext_string}`, so that the "empty string" extension doesn't make the test description look weird